lightning/src/onion_message/messenger.rs

```diff
@@ -192,6 +196,29 @@ impl<Signer: Sign, K: Deref, L: Deref> OnionMessenger<Signer, K, L>
 	}
 }
 
+fn outbound_buffer_full(peer_node_id: &PublicKey, buffer: &HashMap<PublicKey, VecDeque<msgs::OnionMessage>>) -> bool {
+	const MAX_TOTAL_BUFFER_SIZE: usize = (1 << 20) * 128;
+	const MAX_PER_PEER_BUFFER_SIZE: usize = (1 << 10) * 256;
+	let mut total_buffered_bytes = 0;
+	let mut peer_buffered_bytes = 0;
+	for (pk, peer_buf) in buffer {
+		for om in peer_buf {
+			let om_len = om.serialized_length();
```

Comment:
Could we optimize this by implmenting `serialized_length` on `OnionMessage` and `Packet`? Seems a bit wasteful to serialize all pending messages any time an onion message is sent.

Reply:
Ah, right. Was just looking at the implementation and noticed that comment. Feel free to disregard!